### PR TITLE
CircleCI

### PR DESCRIPTION
**Pull request by sgerrand**
💁 Adds CircleCI build configuration to run the following on each formula:
* `brew audit` to test whether they follow Homebrew's guidelines (we may want to push these upstream at some point, so it's helpful if they can be used "as is")
* `brew install` to test that the installation process works

### CODE REVIEW

**Comment by sgerrand** [comment]
```
fc4:
  * Stable: version 2019-07-26_1982 should not end with an underline and a number
Error: 1 problem in 1 formula detected
```
I wonder whether there is scope to change the [format of the releases created by `fc4-framework`](https://github.com/FundingCircle/fc4-framework/blob/1ec0fa03b706bf0bff9fe514eb50236396edc97a/.circleci/config.yml#L125) to use a dash instead of an underscore in the version. 🤔

**Comment by aviflax** [comment]
> I wonder whether there is scope to change the [format of the releases created by `fc4-framework`](https://github.com/FundingCircle/fc4-framework/blob/1ec0fa03b706bf0bff9fe514eb50236396edc97a/.circleci/config.yml#L125) to use a dash instead of an underscore in the version. 🤔

Yes, I’d like to do this. I can bundle it with another change I’d like to make to the version/release identifiers: now that they’re being used for something, I’d like to replace the build number segment (the last segment) with the (short) commit ID (hash) instead. That’s a much more meaningful unit of information. I’ve created https://github.com/FundingCircle/fc4-framework/issues/193 to track this.

That said, I’m thinking we can go ahead and merge this now, even before that change is made to the framework; hopefully the failing builds will motivate us to do that work. (And hopefully they won’t lead to something akin to “broken window syndrome”.)

**Comment by aviflax** [comment]
Do you know whether the install job also tests the formulae (via the call to `test` in the formulae)? If not I’m thinking maybe we should add that.

**Comment by aviflax** [comment]
I ran `brew test fc4` locally and got this:

```shell
$ brew test fc4
Testing fundingcircle/floss/fc4
==> Downloading https://raw.githubusercontent.com/FundingCircle/fc4-framework/c582061ef9d5ad6d680eed71cd55bb2d1e8c3230/docs/tool/fc4-tool-01-context.ya
Already downloaded: /Users/avi.flax/Library/Caches/Homebrew/downloads/89a417ad273c9d1000f701da026472c73c2c0cf651a44de31f2970ccb5cca629--fc4-tool-01-context.yaml
==> /usr/local/Cellar/fc4/2019-07-26_1982/bin/fc4 -fs /private/tmp/fc4-test-20190814-54004-1bqlalm/fc4-tool-01-context.yaml
$ echo $?
0
```

Then I took a look at the output of the install CI job ([build #7](https://app.circleci.com/jobs/github/FundingCircle/homebrew-floss/7)) and I don’t see that output in there anywhere.

So, I’m not entirely sure, but I suspect it’d be good for us to add a post-install invocation of `brew test` for each formula.

**Comment by sgerrand** [comment]
> Do you know whether the install job also tests the formulae (via the call to test in the formulae)?

It doesn't. I'll add it and move the `brew audit` change into a separate branch.

**Comment by sgerrand** [comment]
> I’d suggest maybe commenting it out for now, in this branch, along with a comment explaining why it’s commented out

After consideration, that `audit` job doesn't really need to be commented out as the jobs defined in the workflow dictate what gets run. That job can be safely left inside the current configuration without causing issues.

What do you think?

**Comment by aviflax** [comment]
I agree that that works in the technical sense, but that wasn’t the motivation for my suggestion — which I should have explained. I think that anyone coming across the file in its current state would be confused by why the job is defined but not included in the workflow. So the important part of my suggestion is to add a comment explaining why the audit job is not currently being used. As for commenting out the job itself, I can see some reasons to go either way, and it doesn’t matter much either way.

Sorry I wasn’t clearer in the first place; I hope that makes more sense now.

**Comment by sgerrand** [comment]
Done, thanks! ♻️

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  ci:
+    jobs:
+      # These jobs will fail until the format of fc4-tool release names
+      # is changed to remove underscores in the version component.
+      # This change is being tracked in:
+      # https://github.com/FundingCircle/fc4-framework/issues/193
+      #- audit:
+      #    name: "audit-macos-high-sierra"
+      #    version: "10.2.3"
+      #- test:
```
[the remark this answers is not in the document]
Comment (sgerrand) [reply .circleci/config.yml]: Good catch, thanks! 🔎

**Pull request merged**